### PR TITLE
perf: Replace un-needed context fetching 

### PR DESCRIPTION
**Pull request by sean-brydon**
This PR replaces some fields that were always fetched in trpc session contect to only why they are needed. 

These often made this query exponentially slower whenever there was values in 
* credentials 
* avatar

### CODE REVIEW

File: packages/trpc/server/middlewares/sessionMiddleware.ts [hunk 1]
```diff
@@ -122,7 +108,6 @@ export async function getUserFromSession(ctx: TRPCContextInner, session: Maybe<S
       metadata: orgMetadata,
     },
     id,
-    rawAvatar,
```
Comment (emrysal) [inline packages/trpc/server/middlewares/sessionMiddleware.ts:125]: rawAvatar has a bit more dependencies in the code - see L40 😱

**Review by emrysal — CHANGES_REQUESTED** [review]

File: apps/web/pages/event-types/index.tsx [hunk 2]
```diff
@@ -715,10 +715,7 @@ const EventTypeListHeading = ({
       <Avatar
         alt={profile?.name || ""}
         href={teamId ? `/settings/teams/${teamId}/profile` : "/settings/my-account/profile"}
-        imageSrc={
-          `${orgBranding?.fullDomain ?? WEBAPP_URL}/${teamId ? "team/" : ""}${profile.slug}/avatar.png` ||
-          undefined
-        }
+        imageSrc={`${orgBranding?.fullDomain ?? WEBAPP_URL}/${profile.slug}/avatar.png` || undefined}
```
Comment (emrysal) [inline apps/web/pages/event-types/index.tsx:718]: Fixed a 404, team avatar here was generated as /team/team/seeded-team/avatar.png

File: packages/trpc/server/routers/viewer/eventTypes/create.handler.ts [hunk 3]
```diff
@@ -41,7 +41,23 @@ export const createHandler = async ({ ctx, input }: CreateOptions) => {
...
+        key: true,
+        userId: true,
+        appId: true,
+        invalid: true,
+        teamId: true,
+      },
+      orderBy: {
+        id: "asc",
+      },
+    });
```
Comment (zomars) [inline packages/trpc/server/routers/viewer/eventTypes/create.handler.ts:60]: Would love to abstract these repeated calls. Not a blocker tho.

**Review by zomars — APPROVED** [review]
Nice improvements @sean-brydon :pray:

File: packages/trpc/server/routers/viewer/eventTypes/create.handler.ts (hunk 3 above)
Comment (sean-brydon) [reply packages/trpc/server/routers/viewer/eventTypes/create.handler.ts:60]: I can do this :)

**Review by zomars — DISMISSED** [review]
Checking locally and it seems like we're missing credentials in quite a lot of places. Will block to check. I'll try to see if we can create a "authedProcedureWithCredentials" to use it where needed.

**Review by emrysal — APPROVED** [review]
Should be good to go if tests pass 🙏

**Pull request merged**